packages/video_player_videohole/tizen/src/video_player.cc

```diff
@@ -0,0 +1,510 @@
+}
+
+VideoPlayer::VideoPlayer(FlutterDesktopPluginRegistrarRef registrar_ref,
+                         const CreateMessage &create_message)
+    : registrar_ref_(registrar_ref) {
+  ParseCreateMessage(create_message);
+}
+
+bool VideoPlayer::Open(const std::string &uri) {
+  int ret = player_create(&player_);
+  if (ret != PLAYER_ERROR_NONE) {
+    LOG_ERROR("fail to create media player: %s", get_error_message(ret));
+    return false;
+  }
+
+  if (drm_type_ != DRM_TYPE_NONE && !license_url_.empty()) {
+    drm_manager_ =
+        std::make_unique<DrmManager>(drm_type_, license_url_, player_);
+    drm_manager_->InitializeDrmSession(uri_);
+  }
+
+  ret = player_set_uri(player_, uri.c_str());
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("set uri(%s) failed", get_error_message(ret));
+    return false;
+  }
+  return true;
+}
+
+bool VideoPlayer::SetDisplay(FlutterDesktopPluginRegistrarRef registrar_ref) {
+  int w, h = 0;
+  int ret;
+  if (system_info_get_platform_int("http://tizen.org/feature/screen.width",
+                                   &w) != SYSTEM_INFO_ERROR_NONE ||
+      system_info_get_platform_int("http://tizen.org/feature/screen.height",
+                                   &h) != SYSTEM_INFO_ERROR_NONE) {
+    LOG_ERROR("could not obtain the screen size.");
+    return false;
+  }
+
+  FlutterDesktopViewRef view_ref =
+      FlutterDesktopPluginRegistrarGetView(registrar_ref);
+  if (view_ref == nullptr) {
+    LOG_ERROR("could not get window view handle");
+    return false;
+  }
+
+  void *libHandle = dlopen("libcapi-media-player.so.0", RTLD_LAZY);
+  int (*player_set_ecore_wl_display)(
+      player_h player, player_display_type_e type, void *ecore_wl_window, int x,
+      int y, int width, int height);
+  if (libHandle) {
+    *(void **)(&player_set_ecore_wl_display) =
+        dlsym(libHandle, "player_set_ecore_wl_display");
+    if (player_set_ecore_wl_display) {
+      ret = player_set_ecore_wl_display(
+          player_, PLAYER_DISPLAY_TYPE_OVERLAY,
+          FlutterDesktopViewGetNativeHandle(view_ref), 0, 0, w, h);
+    } else {
+      LOG_ERROR("[VideoPlayer] Symbol not found: %s ", dlerror());
+    }
+    dlclose(libHandle);
+  } else {
+    LOG_ERROR("[VideoPlayer] dlopen failed: %s ", dlerror());
+  }
+
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("[VideoPlayer] player_set_ecore_wl_display failed: %s",
+              get_error_message(ret));
+    return false;
+  }
+  ret = player_set_display_mode(player_, PLAYER_DISPLAY_MODE_DST_ROI);
+  if (ret != PLAYER_ERROR_NONE) {
+    player_destroy(player_);
+    LOG_ERROR("fail to set display mode :%s", get_error_message(ret));
+    return false;
+  }
+  return true;
+}
+
+int64_t VideoPlayer::Create() {
+  int ret;
+  player_id_ = gPlayerIndex++;
+  if (uri_.empty()) {
+    LOG_ERROR("uri is empty");
+    return -1;
+  }
+
+  if (!Open(uri_)) {
+    LOG_ERROR("open failed, uri : %s", uri_.c_str());
```

Comment:
The tag `[VideoPlayer]` does not exist. Please clean up all logs starting with LOG_ so that they have the same format.

Reply:
> The tag `[VideoPlayer]` does not exist. Please clean up all logs starting with LOG_ so that they have the same format.

have fixed